src/Model/Process.php

```diff
@@ -5,68 +5,310 @@
 namespace Friendica\Model;
 
 use Friendica\BaseObject;
+use Friendica\Core\Lock;
+use Friendica\Database\DBM;
+use Friendica\Network\HTTPException\InternalServerErrorException;
 use Friendica\Util\DateTimeFormat;
 use dba;
 
+use Exception;
+
 require_once 'include/dba.php';
 
 /**
- * @brief functions for interacting with a process
+ * @brief functions for interacting with processes
  */
 class Process extends BaseObject
 {
+	const PROC_APP = 'app';
+	const PROC_WORKER = 'wrk';
+	const PROC_DAEMON = 'daem';
+	const PROC_CONSOLE = 'con';
+
+	/**
+	 * @var Process the current Process
+	 */
+	private static	$instance;
+
+	private $id;
+	private $pid;
+	private $code;
+	private $parent;
+	private $type;
+
+	/**
+	 * Process constructor.
+	 *
+	 * @param int    $id      The global unique id of this process
+	 * @param string $command The command which was executed
+	 * @param string $type    The type of this process (default is frontend-app)
+	 * @param int    $pid     The PID of this process
+	 * @param int    $parent  The parent process of this process
+	 *
+	 */
+	private function __construct($id, $command, $type, $pid, $parent)
+	{
+		$this->id = $id;
+		$this->pid = $pid;
+		$this->parent = $parent;
+		$this->type = $type;
+		$this->code = $this->createCode();
+	}
+
+	/**
+	 * Generates a process code for the logging
+	 *
+	 * @return string a generated process code
+	 */
+	private function createCode()
```

Comment:
Can't this be a `__toString()` magic method?

Reply:
Because we're extensively use the code for logging, is it better to pre-create the code or is PHP fast enough so that the code will get created on demand?

Reply:
I'll admit I'm not sure how to make `__toString()` work with the Singleton design pattern in PHP.